src/ui/React/Documentation.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useState } from "react";
+
+interface Navigator {
+  navigate: (s: string, external: boolean) => void;
+}
+
+export const Navigator = React.createContext<Navigator>({ navigate: () => undefined });
+
+export const useNavigator = (): Navigator => useContext(Navigator);
+
+interface History {
+  pages: string[];
+  page: string;
+  push(p: string): void;
+  pop(): void;
+  home(): void;
+}
+
+const defaultPage = "index.md";
+
+const HistoryContext = React.createContext<History>({
+  page: "",
+  pages: [],
+  push: () => undefined,
+  pop: () => undefined,
+  home: () => undefined,
+});
+
+export const Provider = HistoryContext.Provider;
+export const useHistory = (): History => useContext(HistoryContext);
+
+const onPush = (h: History, p: string): History => {
```

Comment:
Stricter type for onPush
```suggestion
const onPush = (h: History, p: FilePath): History => {
```